File: Programs/Minimum_Equal_Sum_of_Two_Arrays_After_Replacing_Zeros.cpp
class Solution {
public:
    long long minSum(vector<int>& nums1, vector<int>& nums2) {
        const int m = nums1.size(), n = nums2.size();
        
        long long sum1 = 0, sum2 = 0;
        int zero1= 0, zero2= 0;
        for(int i=0; i<m; ++i) {
            if(nums1[i] == 0) ++zero1;
            else sum1 += nums1[i];
        }
        for(int i=0; i<n; ++i) {
            if(nums2[i] == 0) ++zero2;
            else sum2 += nums2[i];
        }

        if(sum1 + zero1 > sum2 + zero2) {
            if(zero2 == 0) return -1;
        } else if(sum1 + zero1 < sum2 + zero2) {
            if(zero1 == 0) return -1;
        }

        long long ans= max(sum1 + zero1, sum2 + zero2);
        return ans;
    }
};

File: Programs/Minimum_Equal_Sum_of_Two_Arrays_After_Replacing_Zeros.py
class Solution:
    def minSum(self, nums1: List[int], nums2: List[int]) -> int:
        sum1= sum2= 0
        zero1= zero2= 0

        for num in nums1:
            if num == 0:
                zero1 += 1
            else:
                sum1 += num
        for num in nums2:
            if num == 0:
                zero2 += 1
            else:
                sum2 += num
        
        if sum1 + zero1 > sum2 + zero2:
            if zero2 == 0:
                return -1
        elif sum1 + zero1 < sum2 + zero2:
            if zero1 == 0:
                return -1
        
        return max(sum1 + zero1, sum2 + zero2)
